File: src/main/java/ru/ratanov/springstart/persistence/repo/FilterRepository.java
package ru.ratanov.springstart.persistence.repo;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import ru.ratanov.springstart.persistence.model.Filter;
import ru.ratanov.springstart.web.Cookies;

import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class FilterRepository {

    private static HashMap<String, String> filters = new HashMap<String, String>() {
        {
            put("t", "Категория");
            put("d", "Год выпуска");
            put("k", "Страна");
            put("w", "Добавлен");
        }
    };

    public static HashMap<String, List<Filter>> getFilters() {

        String BASE_URL = "https://kinozal.guru";
        HashMap<String, List <Filter>> items = new HashMap<>();

        try {
            Document doc = Jsoup
                    .connect(BASE_URL.concat("/top.php"))
                    .cookies(Cookies.getCookies()).get();

            for (Map.Entry<String, String> pair : filters.entrySet()) {
                List<Filter> filterList = new ArrayList<>();
                Element param = doc.selectFirst("select[name=" + pair.getKey() +"]");
                for (Element option : param.select("option")) {
                    Filter filter = new Filter();
                    filter.setKey(pair.getKey());
                    filter.setValue(option.attr("value"));
                    filter.setTitle(option.text());

                    filterList.add(filter);
                }
                items.put(pair.getValue(), filterList);
            }

            return items;
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: src/main/java/ru/ratanov/springstart/persistence/repo/FilterRepository.kt
package ru.ratanov.springstart.persistence.repo

import org.jsoup.Jsoup
import ru.ratanov.springstart.persistence.model.Filter
import ru.ratanov.springstart.persistence.model.Param
import ru.ratanov.springstart.web.Cookies

class FilterRepository {

    companion object {
        private val filters = hashMapOf(
                "t" to "Категория",
                "d" to "Год выпуска",
                "k" to "Страна",
                "w" to "Добавлен"
        )


        fun getFilters(): List<Filter> {
            val BASE_URL = "https://kinozal.guru"
            val items = arrayListOf<Filter>()

            val doc = Jsoup
                    .connect("$BASE_URL/top.php")
                    .cookies(Cookies.getCookies())
                    .get()

            filters.forEach { key, title ->
                val options = doc.selectFirst("select[name=$key]").select("option")
                val params = options.map { Param(it.attr("value"), it.text()) }
                items.add(Filter(title, key, params))
            }

            return items
        }
    }



}